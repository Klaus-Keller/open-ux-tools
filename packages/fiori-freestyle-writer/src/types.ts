import type { OdataService } from '@sap-ux/odata-service-writer';
import type { App, Ui5App } from '@sap-ux/ui5-application-writer';

export enum TemplateType {
    Basic = 'basic',
    Worklist = 'worklist',
    ListDetail = 'listdetail'
}

interface Entity {
    name: string;
    key: string;
    idProperty: string;
    numberProperty?: string;
    unitOfMeasureProperty?: string;
}

export interface BasicAppSettings {
    viewName?: string;
}
export interface WorklistSettings {
    entity: Entity;
}

export interface ListDetailSettings {
    entity: Entity;
    lineItem: Entity;
}

export interface Template<T = {}> {
    type: TemplateType;
    settings: T;
}
export interface FioriApp extends App {
    flpAppId?: string;
}
export interface FreestyleApp<T> extends Ui5App {
    template: Template<T>;
    service?: OdataService;
    app: FioriApp;
}

// We need this for the service version
export { OdataVersion } from '@sap-ux/odata-service-writer';
