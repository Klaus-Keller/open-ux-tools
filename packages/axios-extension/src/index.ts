import type { AxiosError, AxiosRequestConfig } from 'axios';
import { isAxiosError } from './base/odata-request-error';
export * from './abap';
export * from './auth';
export * from './base/odata-service';
export * from './base/service-provider';
export * from './factory';
export { AxiosError, AxiosRequestConfig, isAxiosError };
