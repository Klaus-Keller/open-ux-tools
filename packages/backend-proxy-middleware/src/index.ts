import { createProxy, generateProxyMiddlewareOptions, initI18n } from './base/proxy';

export { generateProxyMiddlewareOptions, createProxy };

initI18n();
