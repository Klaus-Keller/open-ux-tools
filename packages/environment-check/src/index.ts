export { archiveProject } from './archive';
export * from './checks';
export { convertResultsToMarkdown, convertResultsToString, storeResultsZip } from './output';
export * from './types';
