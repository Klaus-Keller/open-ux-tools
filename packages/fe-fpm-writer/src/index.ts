export { generateCustomAction } from './action';
export { CustomAction, TargetControl } from './action/types';
export { enableFPM, FPMConfig } from './app';
export { generateBuildingBlock } from './building-block';
export { BuildingBlockType, Chart, Field, FieldFormatOptions, FilterBar } from './building-block/types';
export { generateCustomColumn } from './column';
export { CustomTableColumn } from './column/types';
export { validateBasePath, validateVersion } from './common/validate';
export { generateControllerExtension } from './controller-extension';
export { ControllerExtension, ControllerExtensionPageType } from './controller-extension/types';
export { generateCustomPage, generateListReport, generateObjectPage } from './page';
export { CustomPage, ListReport, ObjectPage } from './page/types';
export { generateCustomSection } from './section';
export { CustomSection } from './section/types';
export { generateCustomView } from './view';
export { CustomView } from './view/types';
