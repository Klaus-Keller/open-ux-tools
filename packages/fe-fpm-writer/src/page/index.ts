import { validatePageConfig } from './common';
import { generate as generateCustomPage } from './custom';
import { generate as generateListReport } from './list';
import { generate as generateObjectPage } from './object';
import type { CustomPage, ListReport, Navigation, ObjectPage } from './types';

export {
    validatePageConfig,
    generateCustomPage,
    generateObjectPage,
    generateListReport,
    CustomPage,
    ObjectPage,
    ListReport,
    Navigation
};
