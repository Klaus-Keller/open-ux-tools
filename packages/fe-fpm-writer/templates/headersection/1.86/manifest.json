{
    "sap.ui5": {
        "routing": {
            "targets": {
                "<%- target %>": {
                    "options": {
                        "settings": {
                            "content": {
                                "header": {
                                    "facets": {
                                        "<%- name %>": {
                                            "template": "<%- ns %>.<%- name %>",
                                            <%if (typeof position !== 'undefined') {%>"position": {
                                                <% if (position.placement) { %> 
                                                    "placement": "<%- position.placement %>"<% if (position.anchor) { %>,<% } %>
                                                <% } %>
                                                <% if (position.anchor) { %> 
                                                    "anchor": "<%- position.anchor %>" 
                                                <% } %>
                                            },<% } %>
                                            "title": "<%- title %>"
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}