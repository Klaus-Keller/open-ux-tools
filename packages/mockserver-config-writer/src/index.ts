export { t } from './i18n';
export { generateMockserverConfig, removeMockserverConfig } from './mockserver-config';
export { getMockserverConfigQuestions } from './prompt';
export * from './types';
