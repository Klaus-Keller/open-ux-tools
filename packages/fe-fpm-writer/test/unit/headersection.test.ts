import type { Editor } from 'mem-fs-editor';
import { create } from 'mem-fs-editor';
import { create as createStorage } from 'mem-fs';
import { join } from 'path';
import { generateCustomHeaderSection } from '../../src/section';
import { RequestGroupId, type CustomHeaderSection, DesignTime } from '../../src/section/types';
import type { Manifest } from '../../src/common/types';
import { Placement } from '../../src/common/types';
import * as manifestSections from './sample/section/webapp/manifest.json';

const testDir = join(__dirname, 'sample/headersection');

describe('CustomHeaderSection generateCustomHeaderSection', () => {
    let fs: Editor;
    const testVersions = ['1.85', '1.86', '1.98'];

    /*  for custom header sections 2 fragment definitions are supported:
        1. template - standard fragment
        2. templateEdit - fragment in edit mode
        we test 2 scenarios (fragments in same folder and differents folders) for 3 different versions - 1.85, 1.86 and 1.98
    */

    describe('read only and edit fragments in same extension folder', () => {
        let expectedEditFragmentPath: string;

        beforeEach(() => {
            const manifest = JSON.parse(JSON.stringify(manifestSections));
            manifest['sap.ui5'].routing.targets.sample.options.settings.content.body.sections = {};
            fs = create(createStorage());
            fs.delete(testDir);
            fs.write(join(testDir, 'webapp/manifest.json'), JSON.stringify(manifest));
        });

        testVersions.forEach((minUI5Version: string) => {
            test(`for version ${minUI5Version}`, () => {
                const customHeaderSection: CustomHeaderSection = {
                    target: 'sample',
                    name: 'NewCustomHeaderSection',
                    folder: 'extensions/custom',
                    title: 'New Custom Header Section',
                    subTitle: 'Custom Header section subtitle',
                    edit: {
                        name: 'NewCustomHeaderSectionEdit',
                        folder: 'extensions/custom',
                        eventHandler: true
                    },
                    requestGroupId: RequestGroupId.Decoration,
                    flexSettings: {
                        designtime: DesignTime.Default
                    },
                    minUI5Version,
                    eventHandler: true
                };
                const expectedFragmentPath = join(
                    testDir,
                    `webapp/${customHeaderSection.folder}/${customHeaderSection.name}.fragment.xml`
                );
                if (customHeaderSection.edit?.folder) {
                    expectedEditFragmentPath = join(
                        testDir,
                        `webapp/${customHeaderSection.edit.folder}/${customHeaderSection.edit.name}.fragment.xml`
                    );
                }
                generateCustomHeaderSection(testDir, { ...customHeaderSection }, fs);

                const updatedManifest = fs.readJSON(join(testDir, 'webapp/manifest.json')) as Manifest;
                const settings = (
                    updatedManifest['sap.ui5']?.['routing']?.['targets']?.['sample']?.['options'] as Record<string, any>
                )['settings'];
                expect(settings.content).toMatchSnapshot();
                expect(fs.read(expectedFragmentPath)).toMatchSnapshot();
                expect(fs.read(expectedFragmentPath.replace('.fragment.xml', '.js'))).toMatchSnapshot();
                expect(expectedEditFragmentPath).toBeDefined();
                expect(fs.read(expectedEditFragmentPath)).toMatchSnapshot();
                expect(fs.read(expectedEditFragmentPath.replace('.fragment.xml', '.js'))).toMatchSnapshot();
            });
        });
    });

    describe('read only and edit fragments in different extension folders', () => {
        let expectedEditFragmentPath: string;

        beforeEach(() => {
            const manifest = JSON.parse(JSON.stringify(manifestSections));
            manifest['sap.ui5'].routing.targets.sample.options.settings.content.body.sections = {};
            fs = create(createStorage());
            fs.delete(testDir);
            fs.write(join(testDir, 'webapp/manifest.json'), JSON.stringify(manifest));
        });

        // folder in edit property differs
        testVersions.forEach((minUI5Version: string) => {
            test(`for version ${minUI5Version}`, () => {
                const customHeaderSection: CustomHeaderSection = {
                    target: 'sample',
                    name: 'NewCustomHeaderSection',
                    folder: 'extensions/custom',
                    title: 'New Custom Header Section',
                    subTitle: 'Custom Header section subtitle',
                    edit: {
                        name: 'NewCustomHeaderSectionEdit',
                        folder: 'extensions/custom/different',
                        eventHandler: true
                    },
                    requestGroupId: RequestGroupId.Decoration,
                    flexSettings: {
                        designtime: DesignTime.Default
                    },
                    minUI5Version,
                    eventHandler: true
                };
                const expectedFragmentPath = join(
                    testDir,
                    `webapp/${customHeaderSection.folder}/${customHeaderSection.name}.fragment.xml`
                );
                if (customHeaderSection.edit?.folder) {
                    expectedEditFragmentPath = join(
                        testDir,
                        `webapp/${customHeaderSection.edit.folder}/${customHeaderSection.edit.name}.fragment.xml`
                    );
                }
                generateCustomHeaderSection(testDir, { ...customHeaderSection }, fs);

                const updatedManifest = fs.readJSON(join(testDir, 'webapp/manifest.json')) as Manifest;
                const settings = (
                    updatedManifest['sap.ui5']?.['routing']?.['targets']?.['sample']?.['options'] as Record<string, any>
                )['settings'];
                expect(settings.content).toMatchSnapshot();
                expect(fs.read(expectedFragmentPath)).toMatchSnapshot();
                expect(fs.read(expectedFragmentPath.replace('.fragment.xml', '.js'))).toMatchSnapshot();
                expect(expectedEditFragmentPath).toBeDefined();
                expect(fs.read(expectedEditFragmentPath)).toMatchSnapshot();
                expect(fs.read(expectedEditFragmentPath.replace('.fragment.xml', '.js'))).toMatchSnapshot();
            });
        });
    });

    describe('positions', () => {
        // position tests
        const customHeaderSection: CustomHeaderSection = {
            target: 'sample',
            name: 'NewCustomHeaderSection',
            folder: 'extensions/custom',
            title: 'New Custom Header Section',
            subTitle: 'Custom Header section subtitle',
            edit: {
                name: 'NewCustomHeaderSectionEdit',
                folder: 'extensions/custom'
            },
            requestGroupId: RequestGroupId.Decoration,
            flexSettings: {
                designtime: DesignTime.Default
            }
        };
        const positionTests = [
            {
                name: 'Create with anchor - latest ui5',
                position: {
                    placement: Placement.Before,
                    anchor: 'Dummy'
                }
            },
            {
                name: 'Create without anchor - latest ui5',
                position: {
                    placement: Placement.Before
                }
            },
            {
                name: 'Create without position - latest ui5',
                position: undefined
            },
            {
                name: 'Create with anchor - 1.85 ui5',
                position: {
                    placement: Placement.Before,
                    anchor: 'Dummy'
                },
                minUI5Version: '1.85.1'
            },
            {
                name: 'Create without anchor - 1.85 ui5',
                position: {
                    placement: Placement.Before
                },
                minUI5Version: '1.85.1'
            },
            {
                name: 'Create without position - 1.85 ui5',
                position: undefined,
                minUI5Version: '1.85.1'
            }
        ];
        positionTests.forEach((testCase) => {
            test(`Test 'position' property. ${testCase.name}`, () => {
                generateCustomHeaderSection(
                    testDir,
                    {
                        ...customHeaderSection,
                        position: testCase.position,
                        minUI5Version: testCase.minUI5Version
                    },
                    fs
                );
                const manifest = fs.readJSON(join(testDir, 'webapp/manifest.json')) as Manifest;
                const headerSection = (
                    manifest?.['sap.ui5']?.['routing']?.['targets']?.['sample']?.['options'] as Record<string, any>
                )?.['settings']?.['content']?.['header']?.['facets']?.['NewCustomHeaderSection'];
                expect(headerSection).toMatchSnapshot();
            });
        });
    });
});
