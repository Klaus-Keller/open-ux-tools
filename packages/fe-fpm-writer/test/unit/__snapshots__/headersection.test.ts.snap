// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomHeaderSection generateCustomHeaderSection positions Test 'position' property. Create with anchor - 1.85 ui5 1`] = `
Object {
  "name": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "position": Object {
    "anchor": "Dummy",
    "placement": "Before",
  },
  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "templateEdit": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSectionEdit",
  "title": "New Custom Header Section",
  "type": "XMLFragment",
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection positions Test 'position' property. Create with anchor - latest ui5 1`] = `
Object {
  "position": Object {
    "anchor": "Dummy",
    "placement": "Before",
  },
  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "templateEdit": ".NewCustomHeaderSectionEdit",
  "title": "New Custom Header Section",
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection positions Test 'position' property. Create without anchor - 1.85 ui5 1`] = `
Object {
  "name": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "position": Object {
    "anchor": "Dummy",
    "placement": "Before",
  },
  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "templateEdit": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSectionEdit",
  "title": "New Custom Header Section",
  "type": "XMLFragment",
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection positions Test 'position' property. Create without anchor - latest ui5 1`] = `
Object {
  "position": Object {
    "anchor": "Dummy",
    "placement": "Before",
  },
  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "templateEdit": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSectionEdit",
  "title": "New Custom Header Section",
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection positions Test 'position' property. Create without position - 1.85 ui5 1`] = `
Object {
  "name": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "position": Object {
    "anchor": "Dummy",
    "placement": "Before",
  },
  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "templateEdit": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSectionEdit",
  "title": "New Custom Header Section",
  "type": "XMLFragment",
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection positions Test 'position' property. Create without position - latest ui5 1`] = `
Object {
  "position": Object {
    "anchor": "Dummy",
    "placement": "Before",
  },
  "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
  "templateEdit": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSectionEdit",
  "title": "New Custom Header Section",
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.85 1`] = `
Object {
  "body": Object {
    "sections": Object {},
  },
  "header": Object {
    "facets": Object {
      "NewCustomHeaderSection": Object {
        "name": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
        "title": "New Custom Header Section",
        "type": "XMLFragment",
      },
    },
  },
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.85 2`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSection'}\\" text=\\"NewCustomHeaderSection\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.85 3`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.85 4`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/different/NewCustomHeaderSectionEdit'}\\" text=\\"NewCustomHeaderSectionEdit\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.85 5`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.86 1`] = `
Object {
  "body": Object {
    "sections": Object {},
  },
  "header": Object {
    "facets": Object {
      "NewCustomHeaderSection": Object {
        "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
        "templateEdit": ".NewCustomHeaderSectionEdit",
        "title": "New Custom Header Section",
      },
    },
  },
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.86 2`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSection'}\\" text=\\"NewCustomHeaderSection\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.86 3`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.86 4`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/different/NewCustomHeaderSectionEdit'}\\" text=\\"NewCustomHeaderSectionEdit\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.86 5`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.98 1`] = `
Object {
  "body": Object {
    "sections": Object {},
  },
  "header": Object {
    "facets": Object {
      "NewCustomHeaderSection": Object {
        "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
        "templateEdit": ".NewCustomHeaderSectionEdit",
        "title": "New Custom Header Section",
      },
    },
  },
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.98 2`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\" xmlns:macros=\\"sap.fe.macros\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSection'}\\" text=\\"NewCustomHeaderSection\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.98 3`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.98 4`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/different/NewCustomHeaderSectionEdit'}\\" text=\\"NewCustomHeaderSectionEdit\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in different extension folders for version 1.98 5`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.85 1`] = `
Object {
  "body": Object {
    "sections": Object {},
  },
  "header": Object {
    "facets": Object {
      "NewCustomHeaderSection": Object {
        "name": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
        "title": "New Custom Header Section",
        "type": "XMLFragment",
      },
    },
  },
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.85 2`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSection'}\\" text=\\"NewCustomHeaderSection\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.85 3`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.85 4`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSectionEdit'}\\" text=\\"NewCustomHeaderSectionEdit\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.85 5`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.86 1`] = `
Object {
  "body": Object {
    "sections": Object {},
  },
  "header": Object {
    "facets": Object {
      "NewCustomHeaderSection": Object {
        "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
        "templateEdit": ".NewCustomHeaderSectionEdit",
        "title": "New Custom Header Section",
      },
    },
  },
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.86 2`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSection'}\\" text=\\"NewCustomHeaderSection\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.86 3`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.86 4`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSectionEdit'}\\" text=\\"NewCustomHeaderSectionEdit\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.86 5`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.98 1`] = `
Object {
  "body": Object {
    "sections": Object {},
  },
  "header": Object {
    "facets": Object {
      "NewCustomHeaderSection": Object {
        "template": "sapux.fe.fpm.writer.test.extensions.custom.NewCustomHeaderSection",
        "templateEdit": ".NewCustomHeaderSectionEdit",
        "title": "New Custom Header Section",
      },
    },
  },
}
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.98 2`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\" xmlns:macros=\\"sap.fe.macros\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSection'}\\" text=\\"NewCustomHeaderSection\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.98 3`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.98 4`] = `
"<core:FragmentDefinition xmlns:core=\\"sap.ui.core\\" xmlns=\\"sap.m\\">
	<VBox>
		<Button core:require=\\"{ handler: 'sapux/fe/fpm/writer/test/extensions/custom/NewCustomHeaderSectionEdit'}\\" text=\\"NewCustomHeaderSectionEdit\\" press=\\"handler.onPress\\" />
	</VBox>
</core:FragmentDefinition>"
`;

exports[`CustomHeaderSection generateCustomHeaderSection read only and edit fragments in same extension folder for version 1.98 5`] = `
"sap.ui.define([
    \\"sap/m/MessageToast\\"
], function(MessageToast) {
    'use strict';

    return {
        onPress: function(oEvent) {
            MessageToast.show(\\"Custom handler invoked.\\");
        }
    };
});
"
`;
